common/src/main/java/com/wynntils/features/user/overlays/NpcDialogueOverlayFeature.java

```diff
@@ -195,11 +211,14 @@ private void renderDialogue(PoseStack poseStack, String currentDialogue) {
 
             // Render "To continue" message
             List<TextRenderTask> renderTaskList = new LinkedList<>();
-            TextRenderTask pressSneakMessage = new TextRenderTask("§cPress SNEAK to continue", renderSetting);
-            renderTaskList.add(pressSneakMessage);
 
-            if (scheduledAutoProgressKeyPress != null && !scheduledAutoProgressKeyPress.isCancelled()) {
-                long timeUntilProgress = scheduledAutoProgressKeyPress.getDelay(TimeUnit.MILLISECONDS);
+            if (currentlyNeedsConfirmation) {
+                TextRenderTask pressSneakMessage = new TextRenderTask("§cPress SNEAK to continue", renderSetting);
+                renderTaskList.add(pressSneakMessage);
+            }
+
+            if (scheduledAutoProgress != null && !scheduledAutoProgress.isCancelled()) {
+                long timeUntilProgress = scheduledAutoProgress.getDelay(TimeUnit.MILLISECONDS);
                 TextRenderTask autoProgressMessage = new TextRenderTask(
                         ChatFormatting.GREEN + "Auto-progress: "
                                 + Math.max(0, Math.round(timeUntilProgress / 1000f))
```

Comment:
I feel like I'm being a real PITA tonight, but this still does not seem right. :-)

Overall I think this latest iteration makes most sense, there are two "kinds" of things we can schedule, an "press sneak for me automatically" and one "remove this overlay after a while". But. The text you add here applies only to the former, since "Press Y to cancel" does not make sense in the latter situation.

Or, at least, this requires some more thought. What would it mean to cancel? Perhaps you need extra time to read it. But if you cancel it, can you "restart" the timer that will clear the message? How? Or is that just making things complicated?

And if you missed the message, how can you re-read it? I don't think these "direct to chat" messages are stored in the quest book history, so if you're a slow reader, it's just lost.

Reply:
I just had a revelation: Perhaps we can solve all the above problems by this behavior: For a scheduled auto progress for non-need-confirmation messages, the text says "Auto-Progress: .... Press Y to move to chat", and pressing Y will then cancel the NPC Dialogue overlay, but at the same time send out the dialogue to the chat. This way you can either give yourself more time to read it, and/or you can get it out of the way.